core/src/main/scala/org/apache/spark/executor/ProcfsMetricsGetter.scala

```diff
@@ -199,7 +201,7 @@ private[spark] class ProcfsMetricsGetter(procfsDir: String = "/proc/") extends L
       case f: IOException =>
         logWarning("There was a problem with reading" +
           " the stat file of the process. ", f)
-        ProcfsMetrics(0, 0, 0, 0, 0, 0)
+        throw f
```

Comment:
According to the proposal, we want to fail at the first `Exception`? Did I understand correctly?

Reply:
What do you mean by the first Exception? I think we will return `ProcfsMetrics(0, 0, 0, 0, 0, 0)` when the first exception occurs.

Reply:
I meant that we escape from `for (p <- pids) {}` statement at the first exception.

Reply:
Yes, that is correct.